### PR TITLE
Mini-cart: add the option to change the style between 'Outline' and 'Fill' in all the buttons

### LINKED ISSUE #8787: Mini-cart: add the option to change the style between 'Outline' and 'Default' in all the buttons
On the `Button` block it is possible to choose the button style between 'Outline' and 'Default', we should consider adding this possibility to the three Mini Cart buttons (`Start shopping`, `View my cart` and `Go to checkout`).

### PR DESCRIPTION
**Pull request by albarin**
This PR adds the `Fill` & `Outline` styles to the 3 `Mini-cart` buttons.

Fixes https://github.com/woocommerce/woocommerce-blocks/issues/8787

#### User-Facing Testing

1. Add the `Mini cart` block to the header template of your site.
2. Go to the `Site Editor` and edit the `Mini cart` template part.
3. Open the `List view` and select the `Empty Mini Cart view`.
4. Click on the `Start shopping` button, check that you can change the button style between `Fill` and `Outline` in the sidebar settings and change it from the default and save.
6. Go to the frontend and check the button has the style you just picked on the editor.
7. Repeat the same steps for the `Filled Mini Cart view` and the `View my cart` and the `Go to checkout` buttons.

* [ ] Do not include in the Testing Notes 

### WooCommerce Visibility

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental

### Changelog

> Mini-cart: add the option to change the style between 'Outline' and 'Fill' in all the buttons.

### CODE REVIEW

File: assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-cart-button-block/block.json
Comment (imanish003) [inline assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-cart-button-block/block.json:1]: I am unsure if it’s related to this PR, but the size of the buttons is quite large on the frontend. Additionally, the background color of the filled button is different on the frontend compared to the editor.

### Editor
image
image


### Frontend
image
image

File: assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/utils.ts
Comment (imanish003) [inline assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/utils.ts:1]: As the className parameter can be undefined, we could either set a default value for it like this:
```
export const getVariant = (
	className = '',
	defaultVariant: Variant
): Variant => {
```
Or we could check if className is undefined and return the defaultVariant early like this:
```
export const getVariant = (
  defaultVariant: Variant,
  className?: string
): Variant => {
  if (className == undefined) {
    return defaultVariant;
  }
```

What do you think?

**Review by imanish003 — COMMENTED** [review]
Great job, @albarin! 🙌🏻
I have some minor suggestions that I would like to share.

File: assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/utils.ts
Comment (albarin) [reply assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/utils.ts:1]: Good catch, I've pushed the change.

File: assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-cart-button-block/block.json
Comment (albarin) [reply assets/js/blocks/mini-cart/mini-cart-contents/inner-blocks/mini-cart-cart-button-block/block.json:1]: This issue about the button's sizes and color is already known and tracked in this issue: https://github.com/woocommerce/woocommerce-blocks/issues/8587

**Review by imanish003 — APPROVED** [review]
Thanks for adding the default value for className 🙌🏻 LGTM 🚀

**Pull request merged**